metadata-ingestion/src/datahub/ingestion/api/incremental_lineage_helper.py

```diff
@@ -19,6 +21,72 @@
 from datahub.specific.dataset import DatasetPatchBuilder
 
 
+class PatchEntityAspect:
+    SKIPPABLE_ATTRIBUTES: ClassVar[List[str]] = [
+        "ASPECT_INFO",
+        "ASPECT_NAME",
+        "ASPECT_TYPE",
+        "RECORD_SCHEMA",
+    ]
+    aspect: Union[ChartInfoClass, DashboardInfoClass]
+    patch_builder: MetadataPatchProposal
+    attributes: List[str]
+
+    def __init__(
+        self,
+        # The PatchEntityAspect can patch any Aspect, however to silent the lint Union is added for DashboardInfoClass
+        # We can use it with any Aspect
+        aspect: Union[DashboardInfoClass, ChartInfoClass],
+        patch_builder: MetadataPatchProposal,
+    ):
+        self.aspect = aspect
+        self.patch_builder = patch_builder
+        self.attributes = dir(self.aspect)
+
+    def is_attribute_includable(self, attribute_name: str) -> bool:
+        """
+        a child class can override this to add additional attributes to skip while generating patch aspect
+        """
+        if (
+            attribute_name.startswith("__")
+            or attribute_name.startswith("_")
+            or attribute_name in PatchEntityAspect.SKIPPABLE_ATTRIBUTES
+        ):
+            return False
+
+        return True
+
+    def attribute_path(self, attribute_name: str) -> str:
+        """
+        a child class can override this if path is not equal to attribute_name
+        """
+        return f"/{attribute_name}"
+
+    def patch(self) -> Optional[MetadataChangeProposalClass]:
+        # filter property
+        properties = {
+            attr: getattr(self.aspect, attr)
+            for attr in self.attributes
+            if self.is_attribute_includable(attr)
+            and not callable(getattr(self.aspect, attr))
```

Comment:
this is probably too generic, and won't work for things that aren't simple attributes (e.g lists, property dicts). For now, please just add `add_dashboard_title(...)` and similar methods to the DashboardPatchBuilder, and call them from `convert_dashboard_info_to_patch`

I do like the idea though - but given the complexities / edge cases, it will probably make sense to use code generation to make it generic

Reply:
Done